File: groups/group5/Java-Recursion/Euclidean/Euclidean.java
package Euclidean;

public class Euclidean {
    public static String process(String[] inputs) throws Exception {
        int num1 = Integer.parseInt(inputs[0].trim());
        int num2 = Integer.parseInt(inputs[1].trim());
        if (num1 <= 0 || num2 <= 0) throw new Exception("Positive integers only.");
        int a = Math.max(num1, num2), b = Math.min(num1, num2);
        int origA = a, origB = b;
        StringBuilder sb = new StringBuilder("SOLUTION:\n");
        while (b > 0) {
            int q = a / b;
            int r = a % b;
            if (r == 0) sb.append(a).append(" = ").append(b).append(" (").append(q).append(")\n");
            else sb.append(a).append(" = ").append(b).append(" (").append(q).append(") + ").append(r).append("\n");
            a = b;
            b = r;
        }
        sb.append("\nGCD: ").append(a);
        sb.append("\nLCM: ").append((origA * origB) / a);
        return sb.toString();
    }
}

File: groups/group3/Euclidean.js
function main(inputs) {
    const input1 = String(inputs[0] ?? "").trim();
    const input2 = String(inputs[1] ?? "").trim();
    
    // Validate first integer
    if (input1 === '') {
        throw new Error('Please enter the first integer');
    }
    
    if (input1.includes('.')) {
        throw new Error('Decimals are not allowed. Please enter a whole number.');
    }
    
    const a = parseInt(input1);
    
    if (isNaN(a)) {
        throw new Error('Invalid input. Please enter a positive integer.');
    }
    
    if (a <= 0) {
        throw new Error('Please enter a positive integer greater than zero.');
    }
    
    // Validate second integer
    if (input2 === '') {
        throw new Error('Please enter the second integer');
    }
    
    if (input2.includes('.')) {
        throw new Error('Decimals are not allowed. Please enter a whole number.');
    }
    
    const b = parseInt(input2);
    
    if (isNaN(b)) {
        throw new Error('Invalid input. Please enter a positive integer.');
    }
    
    if (b <= 0) {
        throw new Error('Please enter a positive integer greater than zero.');
    }
    
    const m = Math.max(a, b);
    const n = Math.min(a, b);
    
    // Euclidean algorithm with steps
    const steps = [];
    let dividend = m;
    let divisor = n;
    let gcd;
    
    while (true) {
        const q = Math.floor(dividend / divisor);
        const r = dividend % divisor;
        
        if (r === 0) {
            steps.push(`${dividend} = ${divisor}(${q})`);
            gcd = divisor;
            break;
        } else {
            steps.push(`${dividend} = ${divisor}(${q}) + ${r}`);
        }
        
        dividend = divisor;
        divisor = r;
    }
    
    const lcm = Math.floor((m * n) / gcd);
    
    // Format output
    const stepsText = steps.join('\n');
    
    return `✅ SOLUTION

${stepsText}

The integers are: ${m.toLocaleString()} and ${n.toLocaleString()}
The greatest common divisor (GCD) of ${m.toLocaleString()} and ${n.toLocaleString()} is: ${gcd.toLocaleString()}
The least common multiple (LCM) of ${m.toLocaleString()} and ${n.toLocaleString()} is: ${lcm.toLocaleString()}`;
}
